opensrp-app/src/main/java/org/smartregister/service/UserService.java

```diff
@@ -522,13 +525,36 @@ public void saveUserGroup(String userName, String password, LoginResponseData us
         if (keyStore != null && userName != null) {
             try {
                 KeyStore.PrivateKeyEntry privateKeyEntry = createUserKeyPair(userName);
-                if (privateKeyEntry != null && userInfo.team != null && userInfo.team.team != null && userInfo.team.team.uuid != null) {
+
+                if (password == null) {
+                    return;
+                }
+
+
+                String groupId = null;
+
+                SyncConfiguration syncConfiguration = CoreLibrary.getInstance().getSyncConfiguration();
+                if (syncConfiguration.getEncryptionParam() != null) {
+                    SyncFilter syncFilter = syncConfiguration.getEncryptionParam();
+                    if (SyncFilter.TEAM.equals(syncFilter) || SyncFilter.TEAM_ID.equals(syncFilter)) {
+                        groupId = getUserDefaultTeamId(userInfo);
+                    } else if (SyncFilter.LOCATION.equals(syncFilter)) {
+                        groupId = getUserLocationId(userInfo);
+                    } else if (SyncFilter.PROVIDER.equals(syncFilter)) {
+                        groupId = password;
+                    }
+                }
+
+                if (StringUtils.isBlank(groupId)) {
```

Comment:
This would mean that  syncConfiguration.getEncryptionParam() was null so in this case why not use the previous 
`groupId = userInfo.team.team.uuid;`

Reply:
Yes, it should fail if Encryption param is not set. We did not want to endorse one encryption field like the team id. This should be a setting in the application level